embark-anyone.lua

```diff
@@ -0,0 +1,108 @@
+local dialogs = require 'gui.dialogs'
+local choices = {}
+
+function addCivToEmbarkList(info)
+   local viewscreen = dfhack.gui.getDFViewscreen(true)
+
+   viewscreen.start_civ:insert ('#', info.civ)
+   viewscreen.start_civ_nem_num:insert ('#', info.nemeses)
+   viewscreen.start_civ_entpop_num:insert ('#', info.pops)
+   viewscreen.start_civ_site_num:insert ('#', info.sites)
+end
+
+function embarkAnyone()
+   local viewscreen = dfhack.gui.getDFViewscreen(true)
+
+
+   if viewscreen._type ~= df.viewscreen_choose_start_sitest then
+      qerror("This script can only be used on the embark screen!")
+   end
+
+   for i, civ in ipairs (df.global.world.entities.all) do
+      if civ.type == df.historical_entity_type.Civilization then
+         local available = false
+
+         -- Check if civ is already available to embark
+         for _, item in ipairs (viewscreen.start_civ) do
+            if item == civ then
+               available = true
+               break
+            end
+         end
+
+         if not available then
+            local sites = 0
+            local pops = 0
+            local nemeses = 0
+            local histfigs = 0
+            local label = ''
+
+            -- Civs keep links to sites they no longer hold, so check owner
+            -- We also take the opportunity to count population
+            for j, link in ipairs(civ.site_links) do
+               local site = df.global.world.world_data.sites[link.target]
+               if site.civ_id == civ.id then
+                  sites = sites + 1
+
+                  -- DF stores population info as an array of groups of residents (?).
+                  -- Inspecting these further could give a more accurate count.
+                  for _, group in ipairs(site.populace.inhabitants) do
+                     pops = pops + group.count
+                  end
+               end
+
+               -- Count living nemeses
+               for _, nem in ipairs (civ.nemesis_ids) do
+                  if df.global.world.nemesis.all[nem].figure.died_year == -1 then
+                     nemeses = nemeses + 1
+                  end
+               end
+
+               -- Count living histfigs
+               -- Used for death detection. May be redundant.
+               for _, fig in ipairs (civ.histfig_ids) do
+                  if df.global.world.history.figures[fig].died_year == -1 then
```

Comment:
here too. use `df.historical_figure.find(fig)`. I didn't mention this above, but you also have to test the result of `find` for nil, since it's possible that the IDs in the vectors are stale, and the referent has been deleted.